src/WebJobs.Script/Rpc/WebHostLanguageWorkerChannelManager.cs

```diff
@@ -110,10 +122,19 @@ public async Task<bool> ShutdownChannelIfExistsAsync(string language, string wor
             {
                 if (languageWorkerChannels.TryGetValue(workerId, out TaskCompletionSource<ILanguageWorkerChannel> value))
                 {
-                    ILanguageWorkerChannel channel = await value?.Task;
-                    if (channel != null)
+                    try
+                    {
+                        ILanguageWorkerChannel channel = await value?.Task;
```

Comment:
It's a bit strange that we wait for channel initialization to complete before shutting it down. Instead of awaiting, can we `languageWorkerChannels.Remove(workerId)` and then `value.Task.ContinueWith(channel => channel.Dispose())`?

Reply:
Agreed. sounds good. will update